lib/std/Build/Step/Compile.zig

```diff
@@ -594,20 +594,20 @@ pub fn forceUndefinedSymbol(compile: *Compile, symbol_name: []const u8) void {
 
 /// Returns whether the library, executable, or object depends on a particular system library.
 /// Includes transitive dependencies.
-pub fn dependsOnSystemLibrary(compile: *const Compile, name: []const u8) bool {
+pub fn dependsOnSystemLibrary(compile: *Compile, name: []const u8) bool {
     var is_linking_libc = false;
     var is_linking_libcpp = false;
 
     var dep_it = compile.root_module.iterateDependencies(compile, true);
```

Comment:
Why does `iterateDependencies` need to take a mutable Module? I don't think we need to make `compile` a mutable pointer?

Reply:
What do you propose?
If the reverse happened, `@constCast` probably would have solved issue error.

Reply:
I never said a word about `@constCast`. I'm saying that 
- `iterateDependencies` could take a `*const Module`.
- `DependencyIterator.Key` could store a `*const Module`

My point is that it doesn't seem to me that anything in `DependencyIterator` requires it to store a mutable reference to the module, meaning you could work your way backwards and not change `compile` to `*Compile`.